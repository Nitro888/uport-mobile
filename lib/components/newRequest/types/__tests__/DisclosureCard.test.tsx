import React from 'react'
import renderer from 'react-test-renderer'

import ConnectedDisclosureRequest, { DisclosureCard } from '../DisclosureCard'

const DisclosureCardProps = {
  requestId: 'TEST',
  componentId: 'TEST',
  title: 'Test',
  description: 'Test',
  actionButton: {
    disabled: false,
    text: 'Confirm',
    action: () => jest.fn,
  },
  cancelButton: {
    disabled: false,
    text: 'Confirm',
    action: () => jest.fn,
  },
  statsMessage: 'Text',
  requestItems: [],
  appBranding: {
    profileImage: '',
    bannerImage: '',
    requestor: '',
  },
  verifiedCredentials: [],
  missingCredentials: [],
  error: null,
}

describe('Disclosure Request Card', () => {
  it('renders dumb component', () => {
    const tree = renderer.create(<DisclosureCard {...DisclosureCardProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('render connected component', () => {
  //   const globalJest: any = global
  //   const initialState = {}
  //   const store = globalJest.mockStore(initialState)

  //   const wrapper = globalJest
  //     .shallow(<ConnectedDisclosureRequest componentId={'TESTID'} />, { context: { store } })
  //     .toJSON()
  //   expect(wrapper).toMatchSnapshot()
  // })
})
