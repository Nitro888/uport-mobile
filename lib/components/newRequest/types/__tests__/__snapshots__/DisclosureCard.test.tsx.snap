// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure Request Card renders dumb component 1`] = `
Array [
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    contentInsetAdjustmentBehavior="never"
    getTextInputRefs={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={200}
    scrollToBottomOnKBShow={false}
    scrollToInputAdditionalOffset={75}
    startScrolledToBottom={false}
    style={
      Object {
        "backgroundColor": "#EFEEF3",
      }
    }
  >
    <View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#EFEEF3",
            "borderRadius": undefined,
            "flex": 1,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#333333",
                "flex": 1,
                "height": 250,
              }
            }
          >
            <Image
              source=""
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 250,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "borderRadius": undefined,
                  "flex": 1,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": "center",
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": 15,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "overflow": "hidden",
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Image
                  resizeMode="cover"
                  source=""
                  style={
                    Object {
                      "height": 70,
                      "width": 70,
                    }
                  }
                />
              </View>
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": 15,
                    "width": undefined,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                    }
                  }
                >
                  No name provided
                </Text>
              </View>
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#FFFFFF",
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": "#222222",
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": 15,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textAlign": "center",
                  }
                }
              >
                Test
              </Text>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#222222",
                    "borderBottomWidth": 15,
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 20,
                    "borderRadius": undefined,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 20,
                    "borderStyle": "solid",
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": 0,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "transform": Array [
                      Object {
                        "rotate": "180deg",
                      },
                    ],
                    "width": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": "#FFFFFF",
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": 15,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 16,
                  "lineHeight": 22,
                }
              }
            >
              Test
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": 15,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#FFFFFF",
            "borderRadius": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": 15,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#BBBBBB",
                "fontSize": 14,
                "textAlign": "center",
              }
            }
          >
            Text
          </Text>
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": 15,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": 1,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": 15,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 8,
                  "maxWidth": 300,
                  "padding": 15,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": "row",
                    "height": undefined,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#D63A59",
                      "fontSize": 18,
                    }
                  }
                >
                  Confirm
                </Text>
              </View>
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": 2,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#6A54D1",
                  "borderRadius": 8,
                  "maxWidth": 300,
                  "padding": 15,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": "row",
                    "height": undefined,
                    "justifyContent": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": undefined,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                    }
                  }
                >
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
